CadRevealFbxProvider/BatchUtils/ScaffoldPartOptimizers/ScaffoldPartOptimizer.cs

```diff
@@ -0,0 +1,10 @@
+namespace CadRevealFbxProvider.BatchUtils.ScaffoldPartOptimizers;
+
+using CadRevealComposer.Tessellation;
+
+public abstract class ScaffoldPartOptimizer
+{
+    public abstract string GetName();
+    public abstract Mesh[] Optimize(Mesh mesh);
+    public abstract string[] GetPartNameTriggerKeywords();
```

Comment:
I'd consider sending the node as input instead of `GetPartNameTriggerKeywords` (ShouldRunOptimizerOnNode(Node)`) and do matching in the implemented class. Allows for more regex-like matching or matching on mesh attributes in addition to name etc

Reply:
We have agreed to postpone this change since we in the future may not trigger only on part names but rather the geometry of the parts. Moreover, for now we do not require regex like searches.